kb-importer/src/main/java/org/eclipse/steady/kb/Manager.java

```diff
@@ -0,0 +1,235 @@
+      setVulnStatus(vulnId, VulnStatus.NOT_STARTED);
+
+    }
+    BackendConnector backendConnector = BackendConnector.getInstance();
+    for (Path vulnDirPath : vulnDirsPaths) {
+      File vulnDir = vulnDirPath.toFile();
+      String vulnDirStr = vulnDirPath.toString();
+      log.info("Initializing process for directory " + vulnDirPath);
+
+      // It is necessary to copy the arguments to avoid concurrent modification
+      HashMap<String, Object> args = new HashMap<String, Object>(mapCommandOptionValues);
+      args.put(ImportCommand.DIRECTORY_OPTION, vulnDirStr);
+      ImportCommand command = new ImportCommand(this, args, BackendConnector.getInstance());
+      if (mapCommandOptionValues.containsKey(ImportCommand.SEQUENTIAL)) {
+        command.run();
+      } else {
+        executor.submit(command);
+      }
+    }
+    try {
+      executor.shutdown();
+      executor.awaitTermination(24, TimeUnit.HOURS);
+      log.info("Finished importing vulnerabilities");
+    } catch (InterruptedException e) {
+      log.error("Process interrupted");
+      log.error(e.getMessage());
+    }
+    this.isRunningStart = false;
+  }
+
+  private void setUploadConfiguration(HashMap<String, Object> args) {
+    Object uploadConstruct = args.get(ImportCommand.UPLOAD_CONSTRUCT_OPTION);
+    VulasConfiguration.getGlobal()
+        .setProperty(
+            CoreConfiguration.BACKEND_CONNECT,
+            (uploadConstruct != null
+                ? CoreConfiguration.ConnectType.READ_WRITE.toString()
+                : CoreConfiguration.ConnectType.READ_ONLY.toString()));
+  }
+
+  public void kaybeeUpdate() throws IOException, InterruptedException {
+    // TODO : write directory as option/property
+    Process updateProcess = Runtime.getRuntime().exec("/kb-importer/data/kaybee update --force");
+    updateProcess.waitFor();
+    if (updateProcess.exitValue() != 0) {
+      log.error("Failed to update kaybee");
+    }
+  }
+
+  public void kaybeePull() throws IOException, InterruptedException {
+    // TODO : write directories as option/property
+    Process pullProcess = Runtime.getRuntime().exec("/kb-importer/data/kaybee pull -c /kb-importer/conf/kaybeeconf.yaml");
+    pullProcess.waitFor();
+    if (pullProcess.exitValue() != 0) {
+      log.error("Kaybee pull failed");
+    }
+  }
+
+  public void stop() {
+    try {
+      executor.shutdownNow();
+      executor.awaitTermination(24, TimeUnit.HOURS);
+      this.isRunningStart = false;
+    } catch (InterruptedException e) {
+      log.error("Process interrupted");
+      log.error(e.getMessage());
+    }
+  }
+
+  public void importSingleVuln(
+    String vulnDirStr, HashMap<String, Object> mapCommandOptionValues, String vulnId) {
+
+    // Should have a lock per vulnerability
+
+    log.info("Initializing process for directory " + vulnDirStr);
+
+    if (this.executor.isShutdown() || this.executor.isTerminated()) {
+      this.executor = (ThreadPoolExecutor) Executors.newCachedThreadPool();
```

Comment:
why do we need a thread to import a single vulnerability?

Reply:
Indeed, I think it is not necessary